server/plugin/approve.go

```diff
@@ -0,0 +1,74 @@
+package plugin
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+
+	"github.com/nathanaelhoun/mattermost-plugin-circleci/server/circle"
+)
+
+func (p *Plugin) httpHandleApprove(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-Id")
+	circleciToken, exists := p.Store.GetTokenForUser(userID, p.getConfiguration().EncryptionKey)
+
+	if !exists {
+		http.NotFound(w, r)
+	}
+	requestData := model.PostActionIntegrationRequestFromJson(r.Body)
+	if requestData == nil {
+		p.API.LogError("Empty request data")
+		return
+	}
+
+	originalPost, appErr := p.API.GetPost(requestData.PostId)
+	if appErr != nil {
+		p.API.LogError("Unable to get post", "postID", requestData.PostId)
+	} else if _, appErr := p.API.UpdatePost(originalPost); appErr != nil {
+		// TODO : remove the button
+		p.API.LogError("Unable to update post", "postID", originalPost.Id)
+	}
+
+	responsePost := &model.Post{
+		ChannelId: requestData.ChannelId,
+		RootId:    requestData.PostId,
+		UserId:    p.botUserID,
+	}
+
+	workFlowID := fmt.Sprintf("%v", requestData.Context["WorkflowID"])
+	jobs, err := circle.GetWorkflowJobs(circleciToken, fmt.Sprintf("%v", requestData.Context["WorkflowID"]))
```

Comment:
Use the previously computed value
```suggestion
	jobs, err := circle.GetWorkflowJobs(circleciToken, workFlowID)
```